playbooks/appsemblerPlaybooks/update_custom_requirements.yml

```diff
@@ -0,0 +1,106 @@
+- hosts: "edxapp-server"
+  sudo: true
+
+  # Update extra requirements, private requirements, 
+  # collect static assets, and restart edx app and workers
+
+  tasks:
+    - include_vars: roles/common_vars/defaults/main.yml
+    - include_vars: roles/edxapp/defaults/main.yml
+
+
+    - name: install read-only ssh key
+      copy: >
+        content="{{ EDXAPP_GIT_IDENTITY }}" dest={{ edxapp_git_identity }}
+        force=yes owner={{ edxapp_user }} mode=0600
+      when: EDXAPP_USE_GIT_IDENTITY
+
+    - name: set git fetch.prune to ignore deleted remote refs
+      shell: git config --global fetch.prune true
+      sudo_user: "{{ edxapp_user }}"
+
+
+
+    - name: install python private requirements
+      shell: >
```

Comment:
I think you can do this in Ansible 1.9.x w/out shell. See here:

* http://docs.ansible.com/ansible/latest/pip_module.html
and I've only tried this with Ansible 2.x: https://github.com/DjangoBoston/djbox/blob/master/deploy/roles/djsite/tasks/main.yml#L4

As I understand it, the virtualenv is only created if it doesn't exist

Reply:
But I'm interested in what @mrgnr has to say on this

Reply:
@johnbaldwin I actually just copied this from `roles/edxapp/tasks/deploy.yml` but left out some comments from there:

```
  # Need to use shell rather than pip so that we can maintain the context of our current working directory; some
  # requirements are pathed relative to the edx-platform repo. Using the pip from inside the virtual environment implicitly
  # installs everything into that virtual environment.
```
I haven't done any deeper investigation, just taking the author's word for it :)